CHANGELOG/CHANGELOG-2.2.md

```diff
@@ -0,0 +1,164 @@
+
+## Deprecations
+
+Support for the v1beta1 VolumeSnapshot API is deprecated and will be removed in the following release, which will require minimum Kubernetes version 1.20.
+
+## Changes by Kind
+
+### API Change
+
+- Adds support for the CSI `GetCapacityResponse.MaximumVolumeSize` field added in CSI spec v1.4.0. If reported by the CSI driver, that value is also published in the `CSIStorageCapacity` objects. The v1beta1 storage API from Kubernetes >= 1.21 is required. ([#584](https://github.com/kubernetes-csi/external-provisioner/pull/584), [@pohly](https://github.com/pohly))
+
+### Feature
+
+- After volume creation or deletion, those CSIStorageCapacity objects most likely affected by that get refreshed sooner than the others. ([#586](https://github.com/kubernetes-csi/external-provisioner/pull/586), [@pohly](https://github.com/pohly))
+- New metrics data (storage capacity tracking, process and Go runtime, work queues, leaderelection) ([#579](https://github.com/kubernetes-csi/external-provisioner/pull/579), [@pohly](https://github.com/pohly))
+
+### Bug or Regression
+
+- Fix races where external-provisioner may have [stopped provisioning](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/103) for a PVC depending on certain timing conditions (provisioning failed once, next attempt currently running, PVC update arrives from API server). ([#564](https://github.com/kubernetes-csi/external-provisioner/pull/564), [@pohly](https://github.com/pohly))
+- Fixes CSI migration to provision PV objects with in-tree topology labels instead of CSI topology labels for aws-ebs, gce-pd, cinder plugins ([#566](https://github.com/kubernetes-csi/external-provisioner/pull/566), [@Jiawei0227](https://github.com/Jiawei0227))
+- Fixes an issue where redundant CSIStorageCapacity objects might have been created because informers were not handled correctly. ([#590](https://github.com/kubernetes-csi/external-provisioner/pull/590), [@bai3shuo4](https://github.com/bai3shuo4))
+- Fixes race where producing StorageCapacity objects may have failed with `the object has been modified` errors ([#565](https://github.com/kubernetes-csi/external-provisioner/pull/565), [@pohly](https://github.com/pohly))
+- If the in-tree providers are depending on StorageClass for DeleteVolume, volumes could not get deleted after migrating from them. So translate in-tree StorageClass to CSI to get SecretReference in DeleteVolume call for CSI Migration scenario. ([#567](https://github.com/kubernetes-csi/external-provisioner/pull/567), [@huchengze](https://github.com/huchengze))
+
+### Other (Cleanup or Flake)
+
+- Removed redundant log lines for CreateVolume and GetCapacity requests. ([#604](https://github.com/kubernetes-csi/external-provisioner/pull/604), [@chrishenzie](https://github.com/chrishenzie))
```

Comment:
Can you also update this to match the wording of the one in 2.1?

Reply:
Done